lib/diagnostic.js

```diff
@@ -8,31 +8,35 @@
 // Question 1
 // Write a callback for reduce that selects
 // the minimum value in the array
-const min = function min() {
-
+const min = function min(prev, curr) {
+  if (curr < prev){
+    return curr;
+  } else {
+    return prev;
+  }
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
-const sum = function sum() {
-
+const sum = function sum(prev, curr) {
+  return prev + curr;
 };
 
 // Question 3
 // Write a function that returns true
 // if invoking the callback for any element
 // in the array returns a truthy value
 const any = function any(array, callback) {
-
+  return array.any(element => callback(currentValue, index, array));
 };
```

Comment:
Your thinking was on the right track, but this was just meant to be a function iterating over an array.

```
const any = function any(array, callback) {
  for (let i = 0; i < array.length; i++) {
    if (callback(array[i], i, array)) {
      return true;
    }
  }

  return false;
};
```